### ISSUE TITLE
Create a CHANGELOG.md file in root of nfEngine

### ISSUE THREAD

**Issue by lookeypl** [nfprojects/nfengine#45]
As in title. Such file will help us track all issues done in between nfEngine releases on master branch.
- Make sure it will be written in Markdown.
- See if it can be automated by adding a script.

**Comment by mkulagowski** [comment]
I'm currently working on it.
Got some ideas from:
&bull;[changelog-from-git-log](https://coderwall.com/p/5cv5lg/generate-your-changelogs-with-git-log)
&bull;[wikipedia-Markdow](http://en.wikipedia.org/wiki/Markdown)
&bull;[git-log-manual](http://git-scm.com/docs/git-log)

Not sure what is the right format and what branch should it track.
I'm thinking about:
2015-04-17 &bull; [view commit](https://github.com/nfprojects/nfengine/issues/45) &bull; Create a CHANGELOG.md file in root of nfEngine
2015-04-16 &bull; [view commit](https://github.com/nfprojects/nfengine/issues/113) &bull; Implement SSE/AVX code for Math module
2015-03-27 &bull; [view commit](https://github.com/nfprojects/nfengine/issues/114) &bull; Implement Library module in nfCommon
.
.
.

Also as for branch...I'd track devel?

**Comment by mkulagowski** [comment]
This issue should be left for now, so I'll leave here all, that I've gathered:
`git log --pretty=format:'%cd &bull; [view commit](http://github.com/nfprojects/nfengine/commit/%H) &bull; %s' --date=short >> CHANGELOG.md`
This line should suffice.
Remember about that:
`git log master` - lists log from branch _master_
`git log --no-merges` - lists log without merge commits

**Comment by lookeypl** [comment]
Indeed, this should be left. The issue is low-priority, because I planned to add it somewhere right before our first release (which will be... long time from now).

Thanks for the info, though. It will surely come in handy. For now, please focus on higher priority tasks. ;)
